app/controllers/auth_controller.rb

```diff
@@ -4,30 +4,40 @@ class AuthController < ApplicationController
   THIRD_PARTY_AUTH_METHODS = [:okta, :facebook, :google, :apple_auth, :apple_refresh]
 
   def create
+    message = []
     method = THIRD_PARTY_AUTH_METHODS.detect { |method|
       data_attrs[:"#{method}_access_token"] || data_attrs[:"#{method}_id_token"] ||
         data_attrs["#{method}_code"] || data_attrs["#{method}_token"]
     }
     token = case method
     when :apple_auth
       # special case for apple, which has given and family name passed in
-      user, apple_refresh_token = AppleAuthService.find_user_by_auth_code(data_attrs[:apple_auth_code], data_attrs[:apple_given_name], data_attrs[:apple_family_name])
+      user, message, apple_refresh_token = AppleAuthService.find_user_by_auth_code(data_attrs[:apple_auth_code], data_attrs[:apple_given_name], data_attrs[:apple_family_name], data_attrs[:create_user])
```

Comment:
maybe instead of returning an optional error message from the various auth functions, instead raise a custom error `UserNotFound` or `UserAlreadyExists` and then rescue that error to produce the error responses

Reply:
ok, is a good idea to implement

Reply:
implemented custom errors